sdk/core/azure-core-amqp/src/amqp/message_sender.cpp

```diff
@@ -168,6 +173,11 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
 
       m_link.reset();
     }
+
+    if (m_messageSender)
+    {
+      m_messageSender.reset();
+    }
```

Comment:
General question for all methods in the destructor.  Should be specifying nothrow on them to ensure that nothing we do in the destructor throws?

Reply:
Based on my reading, as of C++ 11, all destructors appear to be nothrow unless otherwise specified.